### PR TITLE
Cache Descriptor Set Path 

### PR DESCRIPTION
**Pull request by ifadams**
Hey Folks,

This is a draft PR for the descriptor set path caching.

Currently, to the best of my knowledge, we can't actually _delete_ a descriptor set after creation so this is a fairly minimal change.

It relies on the fact that a descriptor set path, as well as its dimensionality, will never change after creation. This lets us bypass the database query after the initial caching of the path and dimensions for a named set.

### CODE REVIEW

File: src/RSCommand.h [hunk 1]
```diff
@@ -34,13 +34,16 @@
...
 
 #include "PMGDQuery.h"
 #include "queryMessage.pb.h"
 
 // Json parsing files
 #include <jsoncpp/json/value.h>
 
+
+
 namespace VDMS {
```
Comment (cwlacewe) [inline src/RSCommand.h:47]:
```suggestion
#include <jsoncpp/json/value.h>

namespace VDMS {
```

**Review by cwlacewe — DISMISSED** [review]
Looks good to me. Minor removal of extra lines requested.

**Review by s-gobriel — COMMENTED** [review]
Do you think it is necessary to also clear the cache when the index is deleted in
 
https://github.com/IntelLabs/vdms/blob/4eaa75030f45836bb9b5dc5775601148086e5a41/src/vcl/DescriptorSet.cc#L94

**Comment by ifadams** [comment]
> Do you think it is necessary to also clear the cache when the index is deleted in
> 
> https://github.com/IntelLabs/vdms/blob/4eaa75030f45836bb9b5dc5775601148086e5a41/src/vcl/DescriptorSet.cc#L94

This is a concern of mine longer term, but it doesn't look like the "delete" is actually externally connected at the moment. i.e. I dont see any VDMS level logic that lets us fully remove an index. At most, with some gymnastics we _might_ be able to remove the descriptor set reference from the graph database, but even thats a little murky.

@s-gobriel do you know if there's an example or call we can check to see if a client can actually trigger a delete of either the index or the referencing PMGD node?

**Review by s-gobriel — APPROVED** [review]
You are correct @ifadams. I checked the whole source and the memory resident index is never deleted,  so I don't see a problem with the cached path.

**Review by cwlacewe — APPROVED** [review]

File: src/RSCommand.h [hunk 2]
```diff
@@ -49,6 +50,10 @@ class RSCommand {
   const std::string _cmd_name;
   std::map<std::string, int> _valid_params_map;
 
+  static tbb::concurrent_unordered_map<std::string, std::string>
```
[the remark this answers is not in the document]
Comment (ifadams) [reply src/RSCommand.h:53]: At the time there was a specific reason (actually did the dev work weeks ago) and was having some compilation issues with it in the subclass.

I can't for the life of me remember the specifics though.... if its causing some heartburn I can try moving it back to the subclass. There's some vaguely potential value of having it in the super class for more general access for some hand-wavey other future dev but from a functional standpoint right now, meh

**Review by s-gobriel — APPROVED** [review]

**Pull request merged**